File: problem3/solution.js
(function() {
  var largestPrimeFactorOf;
  largestPrimeFactorOf = function(limit, factor) {
    if (factor == null) {
      factor = 2;
    }
    while (limit > 1) {
      if (limit % factor === 0) {
        limit /= factor;
      } else {
        factor += 1;
      }
    }
    return factor;
  };
  console.log(largestPrimeFactorOf(600851475143));
}).call(this);

File: problem3/solution.clj
(defn largest-prime-factor[limit factor]
  (if (== limit factor)
   limit 
    (if (zero? (mod limit factor))
      (largest-prime-factor (/ limit factor) factor)
      (largest-prime-factor limit (inc factor)))))

(println (largest-prime-factor 600851475143 2))
